realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -57,6 +59,8 @@ JNIEXPORT void JNICALL Java_io_realm_internal_SharedRealm_nativeInit(JNIEnv* env
 {
     TR_ENTER()
 
+    hack_init();
```

Comment:
`__attribute__((constructor));` looks much better, but it seems to be a gcc extension, i am not quite sure if clang support it. I will check and replace the `JNI_onLoad()` if it supported for clang as well.